File: features/snippets/external_storage/gcs_setup/gcs_driver_create.ts
import { Storage } from '@google-cloud/storage';
import { GcsStorageDriver } from '@temporalio/external-storage-gcs';
import { GoogleCloudGcsStorageDriverClient } from '@temporalio/external-storage-gcs-google-sdk';
/* eslint-disable @typescript-eslint/no-unused-vars */

{
  // @@@SNIPSTART typescript-gcs-driver-create
  const storage = new Storage();

  const driver = new GcsStorageDriver({
    client: new GoogleCloudGcsStorageDriverClient(storage),
    bucket: 'my-temporal-payloads',
  });
  // @@@SNIPEND
}

File: features/snippets/external_storage/gcs_setup/gcs_driver_create.go
package gcssetup

import (
	"context"
	"log"

	"cloud.google.com/go/storage"
	"go.temporal.io/sdk/contrib/gcp/gcsdriver"
	"go.temporal.io/sdk/contrib/gcp/gcsdriver/gcssdk"
	"go.temporal.io/sdk/converter"
)

func CreateGCSDriver() converter.StorageDriver {
	// @@@SNIPSTART go-gcs-driver-create
	gcsClient, err := storage.NewClient(context.Background())
	if err != nil {
		log.Fatalf("create GCS client: %v", err)
	}

	driver, err := gcsdriver.NewDriver(gcsdriver.Options{
		Client: gcssdk.NewClient(gcsClient),
		Bucket: gcsdriver.StaticBucket("my-temporal-payloads"),
	})
	if err != nil {
		log.Fatalf("create GCS driver: %v", err)
	}
	// @@@SNIPEND
	return driver
}
